grive/src/main.cc

```diff
@@ -206,10 +232,44 @@ int Main( int argc, char **argv )
 		token.Auth( code ) ;
 		
 		// save to config
+		config.Set( "client_secret", Val( secret ) ) ;
+		config.Set( "client_id", Val( id ) ) ;
 		config.Set( "refresh_token", Val( token.RefreshToken() ) ) ;
 		config.Save() ;
 	}
+	else
+	{
+		if( vm.count( "id" ) || vm.count( "secret" ))
+		{
+			Log(
+				"'id' and 'secret' are only valid when grive is run with the \"-a\" option to (re)authenticate",
+				log::critical ) ;
 	
+			return -1;
+		}
+	}
+
+	// This is not included in the below 'refresh-token' try
```

Comment:
I've a commit locally where I put that into the same try block, but that would error on upgrade because teh .grive file would not include a already saved client id/secret. -> Not backward compatible. It would save 10 lines of duplicatec code, though...